### ISSUE TITLE
Custom non-resource route

### ISSUE THREAD

**Issue by llhilton** [cloudcreativity/laravel-json-api#268]
We need a '/me' route for our application that returns the currently-authenticated-via-token user. While it's not a standard resource route, it needs to return the user's information following the JSONAPI specification for a read endpoint. 

I've tried using Guzzle, with or without the json_api helper, but I can't get passed being "unauthorized," even if I generate an access token and added it to the header like so: `'Authorization' => 'Bearer ' . $token,`

Then I tried to simplify. I have a method on my user model that I use to build the for the schema's getAttributes function that looks like so:
```
public function jsonApiAttributes() {
        return $this->camelCaseAttributes(array_except($this->attributesToArray(), array_merge([
            'id',
            'deleted_at',
        ], $this->hidden)));
    }

```
So in the controller function I have set up for this route, I can call that function and return it like so `return response(['data' => $attributes], 200);`. However, this seems to omit any null values and doesn't have the relationships with it. 

https://github.com/cloudcreativity/laravel-json-api/issues/198 seems to indicate that there's a way to have a custom controller that extends the JsonApiController but points to a "me" method that would easily accomplish what I want. However, there's nothing in the issue about what the route would look like, and I found nothing in the documentation, either.

Are any of these three ways the "best" way? If so, how can I take it to completion? If not, how should I do this instead? It FEELS like it should be easy, but I've been tearing my hair out for several hours now with no success!

**Comment by plunkettscott** [comment]
Hey @llhilton,

I took a bit of a different approach to this. We wanted a `/users/@this` route. For this I overloaded the find() method on the adapter so that if $resourceId was equal to @this it did a lookup for the current user's ID, rather than the $resourceId. If the $resourceId was not @this I just called the parent's find() method. This could work if you did `/users/me`or `/users/@me`.

Just another approach.

Scott

**Comment by llhilton** [comment]
I'm stuck with /me, unfortunately. I did manage to figure it (mostly) out. I added `use CreatesResponses, AuthorizesRequests;` as traits to the controller, and I can then just do:
``` public function me(Request $request) {
    	$user = auth()->user();

    	return $this->reply()->content($user);
    }```

Interestingly, the null attributes are still not present in the attributes, but that looks like it's not a problem (and maybe part of the JSON API spec I haven't read yet), so not a problem. Thanks!
